R/antigen.garnish_predict.R

```diff
@@ -554,11 +562,23 @@ nugdt <- lapply(f_mhcnuggets, function(x){
 
         if ("iedb_uuid" %chin% names(dt)){
 
+          # keep blast match that will give most conservative IEDB_A value
+
+          dt[!is.na(iedb_uuid) & effect_type == "IEDB_source",
+              match := Consensus_scores %>% as.numeric %>% min(na.rm = TRUE), by = "iedb_uuid"] %>%
+                .[Consensus_scores == match, match := 0]
+
+          dt <- dt[is.na(match) | match == 0]
+
+          dt[, match := NULL]
+
           data.table::setkey(dt, pep_type, iedb_uuid)
 
           dt[!iedb_uuid %>% is.na,
-             IEDB_A := Consensus_scores[2] /
-               Consensus_scores[1], by = c("iedb_uuid", "MHC")]
+             IEDB_A := (Consensus_scores[2] /
+                           Consensus_scores[1]) *
+               (1 / (1 + (0.0003 * Consensus_scores[2]))),
```

Comment:
can you drop a comment here to explain this line?

Reply:
This is pulled from Greenbaum paper, they apply this correction for large WT kD that would massively inflate DAI value.  It basically boils down to that 0.0003 but there is a more complicated way to do it.  This is just a quick and dirty correction.  Basically it barely affects the DAI value unless the WT consensus_score is very very large.